hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/container/balancer/TestContainerBalancer.java

```diff
@@ -214,15 +215,12 @@ public void unBalancedNodesListShouldBeEmptyWhenClusterIsBalanced() {
     balancerConfiguration.setThreshold(99.99);
     containerBalancer.start(balancerConfiguration);
 
-    // waiting for balance completed.
-    // TODO: this is a temporary implementation for now
-    // modify this after balancer is fully completed
-    try {
-      Thread.sleep(100);
-    } catch (InterruptedException e) { }
+    sleepWhileBalancing(100);
 
     containerBalancer.stop();
+    ContainerBalancerMetrics metrics = containerBalancer.getMetrics();
     Assert.assertEquals(0, containerBalancer.getUnBalancedNodes().size());
+    Assert.assertEquals(0, metrics.getDatanodesNumUnbalanced());
```

Comment:
Can we also assert that unbalanced datanodes and size is some expected number in a simulated test, if it is not very difficult to add?

Reply:
We're already testing this in `testMetrics()` at line 513:
```
    Assert.assertEquals(determineExpectedUnBalancedNodes(
            balancerConfiguration.getThreshold()).size(),
        metrics.getDatanodesNumUnbalanced());
```